scripts/check-new-images-exist.sh

```diff
@@ -35,8 +35,11 @@ for NEW_IMAGE in $NEW_IMAGES; do
   fi
   echo "Checking if image ${SOURCE_IMAGE} exists"
   if ! skopeo inspect --retry-times=3 "docker://${SOURCE_IMAGE}" >/dev/null; then
```

Comment:
Rather than doing a different platform-specific check, I would probably skip platform validation entirely at this step. Just add `--raw` and it will give you the manifest for the gat, without trying to find the current platform in the manifest list.
```suggestion
  if ! skopeo inspect --raw --retry-times=3 "docker://${SOURCE_IMAGE}" >/dev/null; then
```

Reply:
Sounds good, I also tested that out but figured having a log explicitly stating that linux/amd64 couldn't be found would be useful, but I guess the onus is on the PR author to ensure the image has the proper platform / arch they need

Reply:
There is probably additional work necessary once you get past this point though; at the moment we only actually **mirror** linux for `amd64 arm64 arm s390x`:
https://github.com/rancher/image-mirror/blob/7bef50dddab7341cb5ddbc391d87ae8d6175ab04/scripts/image-mirror.sh#L14

I am not honestly sure how this handles windows, probably not correctly:
https://github.com/rancher/image-mirror/blob/7bef50dddab7341cb5ddbc391d87ae8d6175ab04/scripts/image-mirror.sh#L120-L130

Reply:
Yeah, you’re right. 

Windows manifest lists will contain multiple entries for `amd64`, with support for particular windows versions denoted by the `.platform.os.version` field. The current `jq` query calling `reverse` here means that `copy_if_changed` is always detecting changes when there were none, since we don’t differentiate the different `os.versions` in the destination image tags like we do the different `variant` entries. `skopeo copy` still handles this somehow but after checking again it looks like the individual tags created aren’t correct or indicative of the os version. 

I think we would need to take the `.platform.os.version` property into consideration when tagging windows images in the destination repository. That way we can make sure we are comparing the correct digests for each `os.version` and `.platform.architecture` in the same way we compare multiple variants for the same architecture. Would that make sense, or do you think there is a better way forward? I'll try to test this idea out today.

Reply:
Yeah that is what I was thinking as well. We should probably mirror to `${DEST}:${TAG}-${ARCH}-${PLATFORM}` if `${PLATFORM}` is not `linux`, and then handle merging that in the manifest list generation step.